crates/util/src/node_with_source.rs

```diff
@@ -14,6 +18,27 @@ impl<'a> NodeWithSource<'a> {
     pub fn new(node: Node<'a>, source: &'a str) -> Self {
         Self { node, source }
     }
+
+    pub fn children_by_field_id(&self, field_id: u16) -> impl Iterator<Item = Self> + Clone {
+        ChildrenByFieldIterator::new(self, field_id)
+    }
```

Comment:
ideally we add a `named_children_by_field_id` as it comes up so often, could then also get rid of the tree-sitter-util file.

Reply:
Haha, yes, I was already thinking cleaning up `tree_sitter_util` will be my next PR :)